api/src/main/java/com/graphhopper/routing/util/CustomModel.java

```diff
@@ -0,0 +1,209 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.util;
+
+import com.graphhopper.json.geo.JsonFeature;
+import com.graphhopper.util.Parameters;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class is used in combination with CustomProfileConfig.
+ */
+public class CustomModel {
+
+    public static final String KEY = "custom_model";
+
+    static double DEFAULT_D_I = 70;
+    // optional:
+    private Double maxSpeedFallback;
+    private Double headingPenalty = Parameters.Routing.DEFAULT_HEADING_PENALTY;
+    // default value derived from the cost for time e.g. 25€/hour and for distance 0.5€/km, for trucks this is usually larger
+    private double distanceInfluence = DEFAULT_D_I;
+    private Map<String, Object> speedFactor = new HashMap<>();
+    private Map<String, Object> maxSpeed = new HashMap<>();
+    private Map<String, Object> priorityMap = new HashMap<>();
```

Comment:
The names are a bit inconsistent, like `priorityMap` vs `speedFactor` (without 'Map'). For priority the json/yaml field is still 'priority' because of the getter name. Calling `customModel.getPriority()` and getting a map is a bit confusing, but I guess we have to stick to jackson conventions unless we use a mixin again? For now I renamed the fields.

Reply:
Yes, I wanted to avoid a mixin. Thanks.